### PR TITLE
Fix Kokkos::Array<T, 0> default initialization for icpc

### PR DESCRIPTION
**Pull request by nliber**
This should address Issue #7153.

icpc incorrectly errors on `constexpr Kokkos::Array<T, 0>` needing initialization when it has a default constructor, as it is an empty specialization.

Prefer the suggestion of @masterleinad of removing the special member functions to address this issue.

### CODE REVIEW

**Comment by masterleinad** [comment]
@ndellingwood Can you check that this indeed fixes #7153?

File: core/unit_test/TestArrayOps.hpp [hunk 1]
```diff
@@ -16,7 +16,6 @@
 
 #include <gtest/gtest.h>
 #include <Kokkos_Core.hpp>
-#include <numeric>
```
Comment (dalg24) [inline core/unit_test/TestArrayOps.hpp:19]: OK but generally avoid mixing in unrelated changes

**Review by dalg24 — APPROVED** [review]

**Comment by masterleinad** [comment]
It seems the problem is the defaulted constructor, see https://godbolt.org/z/dTv5nTP8e. Can we just remove that instead?

**Review by ndellingwood — APPROVED** [review]
This resolved the compilation errors in the nightly intel/19.0.5 configuration, thanks

**Comment by nliber** [comment]
@masterleinad We'd have to get rid of all the special member functions.

**Comment by dalg24** [comment]
> @masterleinad We'd have to get rid of all the special member functions.

Any good reason not to get rid of them?

**Review by masterleinad — APPROVED** [review]

**Review by ndellingwood — APPROVED** [review]
The most recent changes compiled without issue in the intel/19.0.5 build

**Review by dalg24 — APPROVED** [review]

**Comment by dalg24** [comment]
The SYCL build looks like it is headed for timeout town.  Everything else passed.
